pkg/pool-weighted/contracts/lbp/LBPool.sol

```diff
@@ -283,13 +303,46 @@ contract LBPool is WeightedPool, Ownable2Step, BaseHooks {
     }
 
     function _getPoolSwapEnabled() private view returns (bool) {
-        return _poolState.swapEnabled;
+        // Swaps are only enabled, once the weight change has started.
+        (uint256 startTime, , ) = getGradualWeightUpdateParams();
+        return block.timestamp >= startTime;
     }
 
-    function _setSwapEnabled(bool swapEnabled) private {
-        _poolState.swapEnabled = swapEnabled;
+    function _getTokenSwapAllowed(PoolSwapParams memory params) private view returns (bool) {
+        if (restrictSaleOfBootstrapToken) {
+            if (params.kind == SwapKind.EXACT_IN) {
+                return params.indexIn == _bootstrapTokenIndex ? false : true;
+            } else {
+                // exact out swap
+                return params.indexOut == _bootstrapTokenIndex ? false : true;
+            }
```

Comment:
Hmm I think this is not correct.

You should just check that the index out is not the bootstrap token index. It really doesn't matter which end is fixed and which end is calculated; you only care about not pulling the bootstrap token from the vault.

```suggestion
            // Do not allow pulling bootstrap token out.
            return params.indexOut != _bootstrapTokenIndex;
```